src/dune_rpc_impl/long_poll.ml

```diff
@@ -115,9 +129,22 @@ module Progress = struct
       let remaining = Build_system.Progress.remaining p in
       In_progress { complete; remaining }
 
-  let on_rest_request _poller () = `Delay
+  let init_state () =
+    match Build_system.last_event () with
+    | Some evt -> Some (progress_of_build_event evt)
+    | None -> Some (current ())
 
-  let on_update_inactive _ () = None
+  let reset _ = None
+
+  let on_rest_request _poller = function
+    | None -> `Delay
+    | Some last_event ->
+      if is_transient last_event then
+        `Delay
+      else
+        `Respond last_event
```

Comment:
Does that mean clients will now busy poll as long as the last event is Success/Failed?